applications/GeoMechanicsApplication/custom_elements/steady_state_Pw_piping_element.cpp

```diff
@@ -96,25 +97,61 @@ Initialize(const ProcessInfo& rCurrentProcessInfo)
     KRATOS_CATCH("");
 }
 
+template< unsigned int TDim, unsigned int TNumNodes >
+array_1d<double, TNumNodes> SteadyStatePwPipingElement<TDim, TNumNodes>::HydraulicHeadsFromWaterPressures() {
+
+    const double NumericalLimit = std::numeric_limits<double>::epsilon();
+    const PropertiesType& rProp = this->GetProperties();
+    const GeometryType& rGeom = this->GetGeometry();
+
+    //Defining necessary variables
+    array_1d<double, TNumNodes> NodalHydraulicHead;
+    for (unsigned int node = 0; node < TNumNodes; ++node) {
+        array_1d<double, 3> NodeVolumeAcceleration;
+        noalias(NodeVolumeAcceleration) = rGeom[node].FastGetSolutionStepValue(VOLUME_ACCELERATION, 0);
+        const double g = norm_2(NodeVolumeAcceleration);
+        if (g > NumericalLimit) {
+            const double FluidWeight = g * rProp[DENSITY_WATER];
+
+            array_1d<double, 3> NodeCoordinates;
+            noalias(NodeCoordinates) = rGeom[node].Coordinates();
+            array_1d<double, 3> NodeVolumeAccelerationUnitVector;
+            noalias(NodeVolumeAccelerationUnitVector) = NodeVolumeAcceleration / g;
+
+            const double WaterPressure = rGeom[node].FastGetSolutionStepValue(WATER_PRESSURE);
+            NodalHydraulicHead[node] = -inner_prod(NodeCoordinates, NodeVolumeAccelerationUnitVector)
+                - PORE_PRESSURE_SIGN_FACTOR * WaterPressure / FluidWeight;
```

Comment:
I dont think this is correct above the saturated zone, in the case where the retention model is chosen such that above the saturated zone, the soil is assumed to be unsaturated

Reply:
since, above the saturated zone, the Water pressure is 0. Given this formulation, the head is now defined as the Z coordinate

Reply:
actually, setting the water pressure to 0 above the phreatic line is not implemented. So for now this code works. but we might have to look at it in the future